docs/format.rst

```diff
@@ -330,3 +330,79 @@ The reader load process is as follows:
 2. ``COPY_OBJECT_STORE mydataset.lance/_versions/{version}.manifest-{uuid} mydataset.lance/_versions/{version}.manifest`` reattempt synchronization
 3. ``PUT_EXTERNAL_STORE base_uri, version, mydataset.lance/_versions/{version}.manifest`` update the external store to point to the final manifest
 4. ``RETURN mydataset.lance/_versions/{version}.manifest`` always return the finalized path, return error if synchronization fails
+
+Statistics
+----------
+
+Statistics are stored within Lance files. The statistics can be used to determine
+which pages can be skipped within a query. The null count, lower bound (min),
+and upper bound (max) are stored.
+
+Statistics themselves are stored in Lance's columnar format, which allows for
+selectively reading only relevant stats columns.
+
+Statistic values
+~~~~~~~~~~~~~~~~
+
+Three types of statistics are stored per column: null count, min value, max value.
+The min and max values are stored as their native data types in arrays.
+
+There are special behavior for different data types to account for nulls:
+
+For integer-based data types (including signed and unsigned integers, dates,
+and timestamps), if the min and max are unknown (all values are null), then the
+minimum/maximum representable values should be used instead.
+
+For float data types, if the min and max are unknown, then use ``-Inf`` and ``+Inf``,
+respectively. ``NaN`` values should be ignored for the purpose of min and max
+statistics. If the max value is zero (negative or positive), the max value
+should be recorded as ``+0.0``. Likewise, if the min value is zero (positive
+or negative), it should be recorded as ``-0.0``.
+
+For binary data types, if the min or max are unknown or unrepresentable, then use
+null value. Binary data type bounds can also be truncated. For example, an array
+containing just the value ``"abcd"`` could have a truncated min of
+``"abc"`` and max of ``"abd"``. If there is no truncated value greater than the
+maximum value, then instead use null for the maximum.
+
+.. warning::
+
+    The ``min`` and ``max`` values are not guaranteed to be within the array;
+    they are simply upper and lower bounds. Two common cases where they are not
+    contained in the array is if the min or max original value was deleted and
+    when binary data is truncated. Therefore, statistic should not be used to
+    compute queries such as ``SELECT max(col) FROM table``, except to identify
+    which pages and files definitely do not contain the min or max value.
```

Comment:
I'm not sure you can even use stats alone to determine which files definitely do not contain the min or max value.

In other words, if you have two files and one has min: 0, max: 100 and the other has min: 50, max: 200 and the query is `SELECT max(col) FROM table` then you still have to read both files because the 200 might be deleted and the max may be 100?

Reply:
Ah point taken. I'll remove that.